File: Move Zeros.cpp
// Problem Link: https://leetcode.com/problems/move-zeroes/

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int num=0;
        for(int i=0;i<nums.size();i++)
            num+=(nums[i]==0);
        vector<int> res;
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=0)
                res.push_back(nums[i]);
        }
        while(num--)
            res.push_back(0);
        for(int i=0;i<nums.size();i++)
            nums[i]=res[i];
    }
};

File: Move Zeros.js
//https://leetcode.com/problems/move-zeroes/

/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let num = 0,n=nums.length;
    for(let i=0;i<n;i++) {
        if(nums[i]==0){
            num+=1;
            nums.splice(i,1);
            n-=1;
            i-=1;
        }
    }
    while(num--){
        nums.push(0);
    }
};